presto-native-execution/presto_cpp/main/TaskResource.cpp

```diff
@@ -255,58 +219,57 @@ proxygen::RequestHandler* TaskResource::createOrUpdateBatchTask(
+        VELOX_USER_CHECK_NOT_NULL(updateRequest.fragment);
+
+        auto fragment =
+            velox::encoding::Base64::decode(*updateRequest.fragment);
+        protocol::PlanFragment prestoPlan = json::parse(fragment);
+
+        auto serializedShuffleWriteInfo = batchUpdateRequest.shuffleWriteInfo;
         auto shuffleName = SystemConfig::instance()->shuffleName();
         if (serializedShuffleWriteInfo) {
           VELOX_USER_CHECK(
               !shuffleName.empty(),
               "Shuffle name not provided from 'shuffle.name' property in "
               "config.properties");
         }
-        auto fragment =
-            velox::encoding::Base64::decode(*taskUpdateRequest.fragment);
-        protocol::PlanFragment prestoPlan = json::parse(fragment);
+
         VeloxBatchQueryPlanConverter converter(
-            shuffleName, std::move(serializedShuffleWriteInfo), pool_.get());
-        planFragment = converter.toVeloxQueryPlan(
-            prestoPlan, taskUpdateRequest.tableWriteInfo, taskId);
+            shuffleName, std::move(serializedShuffleWriteInfo), pool_);
+        auto planFragment = converter.toVeloxQueryPlan(
+            prestoPlan, updateRequest.tableWriteInfo, taskId);
+
+        return taskManager_.createOrUpdateBatchTask(
+            taskId, batchUpdateRequest, planFragment);
       });
 }
 
 proxygen::RequestHandler* TaskResource::createOrUpdateTask(
     proxygen::HTTPMessage* message,
     const std::vector<std::string>& pathMatch) {
-  protocol::TaskId taskId = pathMatch[1];
   return createOrUpdateTaskImpl(
       message,
       pathMatch,
-      [this](
-          const protocol::TaskId& taskId,
-          const std::string& updateJson,
-          protocol::TaskUpdateRequest& taskUpdateRequest,
-          velox::core::PlanFragment& planFragment) {
-        taskUpdateRequest = json::parse(updateJson);
-        if (taskUpdateRequest.fragment != nullptr) {
-          auto fragment =
-              velox::encoding::Base64::decode(*taskUpdateRequest.fragment);
-          protocol::PlanFragment prestoPlan = json::parse(fragment);
-          auto converter = VeloxInteractiveQueryPlanConverter(pool_.get());
-          planFragment = converter.toVeloxQueryPlan(
-              prestoPlan, taskUpdateRequest.tableWriteInfo, taskId);
-        }
+      [&](const protocol::TaskId& taskId, const std::string& updateJson) {
+        protocol::TaskUpdateRequest updateRequest = json::parse(updateJson);
+        VELOX_USER_CHECK_NOT_NULL(updateRequest.fragment);
+
+        auto fragment =
+            velox::encoding::Base64::decode(*updateRequest.fragment);
+        protocol::PlanFragment prestoPlan = json::parse(fragment);
+
+        auto converter = VeloxInteractiveQueryPlanConverter(pool_);
+        auto planFragment = converter.toVeloxQueryPlan(
+            prestoPlan, updateRequest.tableWriteInfo, taskId);
+
+        return taskManager_.createOrUpdateTask(
+            taskId, updateRequest, planFragment);
```

Comment:
So, we moved actual task creation from "createOrUpdateTaskImpl" to lambdas in createOrUpdateTask and createOrUpdateBatchTask?
A bit confusing.